pkg/apis/metrics/latest/latest.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package latest
+
+import (
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	"k8s.io/kubernetes/pkg/api/registered"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/apis/metrics"
+	"k8s.io/kubernetes/pkg/apis/metrics/v1alpha1"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/sets"
+)
+
+// ExternalVersions lists all known external versions for this group from most preferred to least preferred
+var ExternalVersions = []unversioned.GroupVersion{v1alpha1.SchemeGroupVersion}
+
+// Codec is the Codec for the most preferred ExternalVersion
+var Codec = v1alpha1.Codec
+
+// RESTMapper is a RESTMapper for all versions of the API group
+var RESTMapper meta.RESTMapper
+
+var Accessor = meta.NewAccessor()
+
+func init() {
+	finalExternalVersions := []unversioned.GroupVersion{}
+
+	for _, allowedVersion := range registered.GroupVersionsForGroup(metrics.SchemeGroupVersion.Group) {
+		for _, externalVersion := range ExternalVersions {
+			if externalVersion == allowedVersion {
+				finalExternalVersions = append(finalExternalVersions, externalVersion)
+			}
+		}
+	}
+
+	ExternalVersions = finalExternalVersions
+	Codec = nil
+	if len(ExternalVersions) > 0 {
+		Codec = runtime.CodecFor(api.Scheme, ExternalVersions[0].String())
+		RESTMapper = newRESTMapper()
+	}
+}
+
+const importPrefix = "k8s.io/kubernetes/pkg/apis/metrics"
+
+func newRESTMapper() meta.RESTMapper {
+	worstToBestGroupVersions := []unversioned.GroupVersion{}
+	for i := len(ExternalVersions) - 1; i >= 0; i-- {
+		worstToBestGroupVersions = append(worstToBestGroupVersions, ExternalVersions[i])
```

Comment:
> I still don't see why we're flipping this list... NewDefaultRESTMapper uses this for two things
> 1. searching for a version if none is specified (so we want it in real priority order)
> 2. adding gvk's to the rest mapper (and first one in wins for resourceToKind, so we want it in real priority order)
> 
> All our API groups have a single version, so it doesn't seem to matter at the moment, but I think this is backwards

Maintains current behavior.  The goal of this pull is avoid changing that any behavior to limit scope.